Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs

```diff
@@ -47,13 +48,38 @@ public override async Task<ProblemResourceAdministrationModel> Edit(ProblemResou
             .Include(pr => pr.Problem)
             .FirstOrDefaultAsync();
 
+        var areFileAndLinkNull = model is { File: null, Link: null };
+
+        if (resource is { File: null } && areFileAndLinkNull)
+        {
```

Comment:
This check seems a bit redundant.. if file and link are null, there is no way of returning something other that true here? So I think only `areFileAndLinkNull` boolean is needed (without the `resource is { File: null }`)

Reply:
If ```resource is { File: null }``` is removed we would not be able to switch from one file-based resource type to another. We would throw the ```BusinessServiceException``` inside of this ```if``` check, if we  want to switch from ```AuthorsSolution``` to ```ProblemDescription``` without ```removing``` the file.